src/Stack/Ghci.hs

```diff
@@ -192,7 +192,7 @@ ghciSetup mainIs stringTargets = do
         forM locals $
         \(name,(cabalfp,components)) ->
              makeGhciPkgInfo sourceMap (map fst locals) name cabalfp components
-    unless (M.null realTargets) (build (const (return ())) Nothing bopts)
+    try (unless (M.null realTargets) (build (const (return ())) Nothing bopts)) >>= (\x -> return (x :: Either SomeException ()))
```

Comment:
Would something like this work?
```
(_ :: Either SomeException ()) <- try (unless (M.null realTargets) (build (const (return ())) Nothing bopts))
```

But catching SomeException isn't great, because that will also hide asynchronous exceptions (like hitting Ctrl-C) that should be allowed to propagate up the call stack.  What about catching `StackBuildException` instead?

@chrisdone, can you review this?  I'm not sure about the implications.  Should the build exception be displayed (maybe as a warning)?